engine/commands/cortex_upd_cmd.h

```diff
@@ -1,13 +1,124 @@
 #pragma once
-#include <string>
 #include <optional>
+#include <string>
+
+#include "httplib.h"
+#include "nlohmann/json.hpp"
+#include "utils/file_manager_utils.h"
+#include "utils/logging_utils.h"
 
 namespace commands {
+#ifndef CORTEX_VARIANT
+#define CORTEX_VARIANT file_manager_utils::kProdVariant
+#endif
+constexpr const auto kNightlyHost = "https://delta.jan.ai";
+constexpr const auto kNightlyFileName = "cortex-nightly.tar.gz";
+const std::string kCortexBinary = "cortex";
+
+inline std::string GetCortexBinary() {
+#if defined(_WIN32)
+  constexpr const bool has_exe = true;
+#else
+  constexpr const bool has_exe = false;
+#endif
+  if (CORTEX_VARIANT == file_manager_utils::kNightlyVariant) {
```

Comment:
Needs test coverage

Reply:
We use CORTEX_VARIANT for different build, this should be added in CI, not unit tests